File: TravelAgencyRestApi/src/main/java/com/travelagency/travelagencyapplication/dto/request/TravelRequest.java
package com.travelagency.travelagencyapplication.dto.request;


import com.travelagency.travelagencyapplication.collection.DayTrip;
import com.travelagency.travelagencyapplication.collection.Pack;
import com.travelagency.travelagencyapplication.collection.User;
import com.travelagency.travelagencyapplication.enums.TransportType;
import com.travelagency.travelagencyapplication.enums.TravelState;
import lombok.*;
import org.springframework.web.multipart.MultipartFile;

import java.time.LocalDate;
import java.util.Date;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TravelRequest {

    private String title;
    private String destination;
    private LocalDate startDate;
    private LocalDate endDate;
    private Set<String> imagesUrl;

    private TransportType transportType;
    private String transportCompany;
    private TravelState travelState;

    private String travelGuideId;
    private Set<PackRequest> packs;
    private Set<TravelDayTripRequest> travelDayTripRequests;
    private Integer maxSeat;
    private Double basedPrice;

    private String shortDescription;
    private String longDescription;

}

File: TravelAgencyFrontend/src/app/types/Travel/TravelRequest.ts
import {TransportType} from "./TransportType";
import {TravelState} from "./TravelState";
import {DayTripResponse} from "../DayTrip/DayTripResponse";
import {Pack} from "../Pack/Pack";
import {TravelDayTripRequest} from "./TravelDayTripRequest";

export interface TravelRequest {
  title: string;
  destination: string;
  startDate: string;
  endDate: string;
  imagesUrl?: string[];
  transportType: TransportType;
  transportCompany: string;
  travelState: TravelState;
  travelGuideId: string;
  packs: Pack[];
  travelDayTripRequests: TravelDayTripRequest[];
  maxSeat: number;
  basedPrice: number;
  shortDescription: string;
  longDescription: string;
}
